### ISSUE TITLE
Save filter and sorting when leaving page

### ISSUE THREAD

**Issue by AlbertServen** [tabalinas/jsgrid#81]
Hi, I want to save the state of the filter and sort when I leave the page and then restore it when I return. It would be nice if there's a method `setFilter` (sets the filter) and `getSorting` (returns the name and order).

Thank you.

**Comment by tabalinas** [comment]
To set filter use `loadData(filter)`. It accepts filtering params.
To get current sorting now you can use `_sortingParams` method. It returns current sorting in the format `{ sortField, sortOrder }`. 
Your suggestions looks reasonable. Thank you!

**Comment by AlbertServen** [comment]
Ok, I will try it on Monday! Thank you very much.

**Comment by AlbertServen** [comment]
I've managed to store the values but when I want to restore the filter, order and page only the order and page stay. I tried using ```jsGrid('loadData')```  and ```jsGrid('search', filter)``` but when I sort (for example) applies an empty filter, it doesn't get stored. Thanks

**Comment by tabalinas** [comment]
How do you restore filter, order and page? Could you provide the code?

**Comment by AlbertServen** [comment]
I've tried different ways (like this and search and sort inverted)
```javascript
grid.jsGrid('search', storedSettings).done(function() {
    grid.jsGrid('sort', sort).done(function() {
        grid.jsGrid('openPage', page);
    }.bind(this));
}.bind(this));
```
without waiting (and also changing orders):
```javascript
grid.jsGrid('search', storedSettings);
grid.jsGrid('sort', sort);
grid.jsGrid('openPage', page);
```
with loadData: (```filter``` contains sortOrder, sortField, pageIndex, etc)
```javascript
grid.jsGrid('loadData', filter);
```

What is wrong?

**Comment by tabalinas** [comment]
Try the following order:

```javascript
grid.search(gridSettings.filter).done(function() {
    grid.sort({
    	field: gridSettings.sorting.sortField,
        order: gridSettings.sorting.sortOrder
    }).done(function() {
        grid.option("pageIndex", gridSettings.pageIndex);
    });
});
```

See the fiddle http://jsfiddle.net/tabalinas/9g8upd33/

**Comment by AlbertServen** [comment]
I've tried it and I get the same result: sorting and page ok but filter empty. If you do ```console.log(grid.getFilter())``` before calling ```grid.sort``` you'll see that the filter isn't the same as ```gridSettings.filter```. I don't know if it is the desired behavior. 
Thanks for all your effort.

**Comment by tabalinas** [comment]
It works in my fiddle, right? Try to find the differences between your case and the fiddle.
Maybe you have loading by page or something else.

**Comment by AlbertServen** [comment]
Yes, I have loading by page, sorry not to mention this. My config:
```javascript
...
autoload: true,
filtering: true,
editing: true,
sorting: true,
paging: true,
pageLoading: true,
pageSize: 20,
...
It could be the reason that isn't working?

**Comment by tabalinas** [comment]
You are right, the reason is `pageLoading: true`. Loading by page strategy calls `loadData` on sorting (since sorting while loading by page should be done externally). `loadData` w/o params loads data with filter from filter row (not the one specified in params).
Thus, you are right. We should have something like `setFilter` and `setInsert` to set values in inputs.
For now you should restore the filter directly with jQuery (select elements by `.jsgrid-filter-row` and set values to inputs). Then you can call `sort` and filter with sorting will be applied.

**Comment by AlbertServen** [comment]
I've done what you said and all is working fine (restoring filter directly with jQuery). Thanks for all your help.

**Comment by tabalinas** [comment]
You are welcome! Thank you for good case to consider and your valuable suggestions!
